check_no_globals.go

```diff
@@ -10,6 +10,38 @@ import (
 	"strings"
 )
 
+func isWhitelisted(i *ast.Ident) bool {
+	return i.Name == "_" || looksLikeError(i)
+}
+
+// looksLikeError checks if the AST identifier starts with 'err' or 'Err'.
+//
+// In https://github.com/leighmcculloch/gochecknoglobals/pull/3 it was
+// decided to not check if the variable really implements the error interface.
+// (This would catch more issues where variables start with 'err' or 'Err' but
+// are not actual errors — but variables of some other type).
+// Doing so will be a bit more complicated as Go tries to infer the type of
+// variables by looking at their initial value. That's great, but in our case
+// it's a bit unfortunate because it means that the AST doesn't necessarily
+// have a variable's type available. See the follwing Go source and example:
+// https://golang.org/src/go/ast/scope.go?s=2224:2289#L80
+//  var (
+//      errNotFound = errors.New("404 not found")       // ident.Obj.Type is nil
+//      errNotFound error = errors.New("404 not found") // ident.Obj.Type is not nil
+//  )
+// If the `error` type information is omitted in a variable's declaration,
+// the AST can't determine the type (that will be done further down below in
+// the compilation toolchain). It also means that all type information would
+// need to be available to the compiler, not just the types declared in the
+// current file. There still is room for improvement here!
```

Comment:
I think it's worth taking these extra lines that talk about how we're limiting the implementation, and putting them into a new issue on the project instead of in this comment. Lets keep this comment to describing what the function does today, but all of these details are super useful to if we're going to improve this, and the best place for that info is an issue where we'll be discussing how to improve it.

Reply:
Done. See https://github.com/leighmcculloch/gochecknoglobals/issues/5